addons/website_payment/models/res_config_settings.py

```diff
@@ -46,14 +48,20 @@ def _compute_acquirers_state(self):
                 config.acquirers_state = 'none'
 
     def action_activate_stripe(self):
+        self.ensure_one()
+        if not self.is_stripe_supported_country:
+            return {
+                'type': 'ir.actions.client',
+                'tag': 'display_notification',
+                'params': {
+                    'type': 'info',
+                    'message': _("Stripe Connect is not available in your country, please use another payment provider."),
+                }
+            }
```

Comment:
Following the comment above, you no longer need to handle this case.
```suggestion
```

Reply:
Good idea. It would also help is someone calls that method in a community module or through RPC, for some reason.